tests/integration_tests/datasource_tests.py

```diff
@@ -26,30 +26,24 @@
 from superset.datasets.commands.exceptions import DatasetNotFoundError
 from superset.exceptions import SupersetException, SupersetGenericDBErrorException
 from superset.utils.core import get_example_database
+from tests.integration_tests.base_tests import db_insert_temp_object, SupersetTestCase
 from tests.integration_tests.fixtures.birth_names_dashboard import (
     load_birth_names_dashboard_with_slices,
 )
-
-from .base_tests import db_insert_temp_object, SupersetTestCase
-from .fixtures.datasource import datasource_post
+from tests.integration_tests.fixtures.datasource import get_datasource_post
 
 
 class TestDatasource(SupersetTestCase):
     def setUp(self):
-        self.original_attrs = {}
```

Comment:
I really have no idea why this logic is here and the `before_update` method exposed its fragility. Rather than using instance variables to save state and restore state, using a sub-transaction and rolling back seems considerably cleaner.